src/kernels/ipywidgets/scriptUriConverter.ts

```diff
@@ -1,122 +1,22 @@
-import { inject, injectable } from 'inversify';
-import { IFileSystem } from '../../platform/common/platform/types';
-import { IExtensionContext, IsWebExtension } from '../../platform/common/types';
-import { sha256 } from 'hash.js';
-import { createDeferred, Deferred } from '../../platform/common/utils/async';
-import { traceInfo, traceError } from '../../platform/logging';
-import { getComparisonKey } from '../../platform/vscode-path/resources';
-import { getFilePath } from '../../platform/common/platform/fs-paths';
-/* eslint-disable @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports */
-const sanitize = require('sanitize-filename');
 
-@injectable()
 export class ScriptUriConverter implements ILocalResourceUriConverter {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    public get requestUri(): Event<Uri> {
-        return this.requestUriEmitter.event;
-    }
-    public get rootScriptFolder(): Uri {
-        return this._rootScriptFolder;
-    }
-    private readonly _rootScriptFolder: Uri;
-    private readonly createTargetWidgetScriptsFolder: Promise<Uri>;
-    private readonly targetWidgetScriptsFolder: Uri;
-    private readonly resourcesMappedToExtensionFolder = new Map<string, Promise<Uri>>();
-    private readonly uriConversionPromises = new Map<string, Deferred<Uri>>();
-    private requestUriEmitter = new EventEmitter<Uri>();
+    constructor(private readonly isWebExtension: boolean, private readonly converter: (input: Uri) => Promise<Uri>) {}
 
     /**
      * This method is called to convert a Uri to a format such that it can be used in a webview.
-     * WebViews only allow files that are part of extension and the same directory where notebook lives.
-     * To ensure widgets can find the js files, we copy the script file to a into the extensionr folder  `tmp/nbextensions`.
-     * (storing files in `tmp/nbextensions` is relatively safe as this folder gets deleted when ever a user updates to a new version of VSC).
-     * Hence we need to copy for every version of the extension.
-     * Copying into global workspace folder would also work, but over time this folder size could grow (in an unmanaged way).
+     * Some times we have resources in the extension/tmp folder that contain static resources such as JS, image files and the like.
+     * The webview can load them, however the Uri needs to be in a special format for that to work.
+     * Note: Currently this is only use for Jupyter Widgets.
      */
-    public async asWebviewUri(localResource: Uri): Promise<Uri> {
-        if (this.isWeb) {
-            // We cannot create folders on the web, return this as is.
-            return localResource;
-        }
-        // Make a copy of the local file if not already in the correct location
-        if (!this.isInScriptPath(localResource)) {
-            const key = getComparisonKey(localResource);
-            if (!this.resourcesMappedToExtensionFolder.has(key)) {
```

Comment:
This is no longer required. We copy the entire nbextensions folders now, instead of copying a single file